### PR TITLE
Update riseup-vpn: 0.21.11 -> 0.24.5-unstable-2024-07-08

### PR DESCRIPTION
**Pull request by vigneshpai2003**
## Description of changes

`riseup-vpn` was updated from the unusable `21.11` version (details in this [issue](https://github.com/NixOS/nixpkgs/issues/321132)) to track the main branch. The package was ported to `qt6`.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by vigneshpai2003** [comment]
The `.qrc` file needs to manually added because `qmake` (located at `qtbase/bin`) searches for the `qmlcachegen` binary in `qtbase/libexec` while the binary is located in `qtdeclarative/libexec`. Hence this particular `.qrc` file cannot be built automatically. If anyone know how to fix this in qt6, please let me know.

For now, the hack of adding the `.qrc` file manually does the job.

**Comment by h4rldev** [comment]
+1

**Review by ju1m — APPROVED** [review]
LGTM, thanks!

**Comment by h4rldev** [comment]
Let's get this merged! :D

**Comment by h4rldev** [comment]
bump

File: pkgs/tools/networking/bitmask-vpn/default.nix [hunk 1]
```diff
@@ -13,32 +13,29 @@
...
+, qtsvg
 , qttools
+, qtwayland
 , CoreFoundation
 , Security
 , provider ? "riseup"
 }:
 let
-  version = "0.21.11";
+  version = "git";
```
Comment (Aleksanaa) [inline pkgs/tools/networking/bitmask-vpn/default.nix:26]: https://github.com/NixOS/nixpkgs/tree/master/pkgs#package-naming

Comment (vigneshpai2003) [reply pkgs/tools/networking/bitmask-vpn/default.nix]: I have updated with the correct naming convention, thanks

**Review by ju1m — APPROVED** [review]

**Comment by h4rldev** [comment]
bump

**Comment by h4rldev** [comment]
bump

**Comment by h7x4** [comment]
The changes look good, but could you tidy up the commits a bit? Assuming this is all part of the package upgrade, maybe a single commit with message `bitmask-vpn: 0.21.11 -> 0.24.5-unstable-2024-07-08` would be good? https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#commit-conventions

**Comment by RossComputerGuy** [comment]
Mass ping & merge conflict. Please reopen your PR with the correct branch.

**Pull request closed without merging**